mood-analysis.rb

```diff
@@ -28,5 +33,45 @@ def analyze_mood(words)
   "05/11 Yay, yay, yay! I'm having a awfuly great day."
 ]
 
-puts analyze_mood(text[0])
-puts analyze_mood(text[1])
+text.each do |element|
+  date = element[0..4]
+  smiley = analyze_mood(element)
+  puts date + " "+ smiley
+end
+
+def happy_days(entries)
+  happy_entry = 0
+  general_entry = 0
+  #keep track of # of happy days and entries for each entry
+  entries.map do |entry|
+    happy_entry += 1 if analyze_mood(entry) == ":-)"
+    general_entry += 1
+    break if happy_entry == 3
+  end
+  #output "it takes 5 entries for 3 happy days to occur"
+  if happy_entry < 3
+    puts "Sorry, not enough happy days!"
+  else
+    puts "It takes #{general_entry} entries for 3 happy days to occur."
+  end
+end
+
+def overall_mood(entries)
+  frequency = {
+    ":-)" => 0,
+    ":-(" => 0,
+    ":-|" => 0
+  }
+
+  entries.map do |entry|
+    mood_instance = analyze_mood(entry)
+    frequency[":-)"] += 1 if mood_instance == ":-)"
+    frequency[":-("] += 1 if mood_instance == ":-("
+    frequency[":-|"] += 1 if mood_instance == ":-|"
+  end
+
+  winner = frequency.max_by{|key, value| value}
+  puts "The most common mood is #{winner[0]}"
+end
```

Comment:
Here I also would have liked to see the smallest piece of information returned. In this case, just the emoticon.